src/shared/Core/Authentication/OAuth/OAuth2Client.cs

```diff
@@ -383,10 +386,10 @@ protected Exception CreateExceptionFromResponse(string json)
         {
             if (TryCreateExceptionFromResponse(json, out OAuth2Exception exception))
             {
-                return exception;
+                return new Trace2OAuth2Exception(_trace2, exception.Message);
             }
 
-            return new OAuth2Exception($"Unknown OAuth error: {json}");
+            return new Trace2OAuth2Exception(_trace2, $"Unknown OAuth error: {json}");

```

Comment:
Does `json` have PII in it? 

It looks like we only get here if the JSON parser barfs on the string, but do we know it is safe to log it?

Reply:
Mmm...probably not. I'll work on adding the format here.